### ISSUE TITLE
[Q] Alternative Hardware

### ISSUE THREAD

**Issue by mdaser** [CWempe/Thermostat-Controller#2]
Hello CWempe,
thanks for the detailed instructions and hints to make the TRQ 21 T smarter and connect it to an MQTT server.

As far as I understood, the relay in your solution  switches the black connector on J1 either to green or yellow on J1; being connected to green if the relay is in idle.

I was wondering, whether I could used a device like this:
https://circuit.rocks/relay-esp8266-10a-220v-network-wifi-module-input-dc-7v-30v.html

It may be powered with the 24V of the thermostat (it accepts 7 - 30V) and run something like Tasmota to connect to an MQTT server.
The MQTT server implements the logic, when switching on or off (or rather from day to night mode and vice versa).

Does that sound reasonable?

Thanks in advance!
     Martin

**Comment by CWempe** [comment]
I think this should work. (NO GARANTEE!!).

But I am not sure if the board will fit in the thermostat.

Here is a photo from my setup.
![20170410_212205](https://user-images.githubusercontent.com/1270359/96354459-3fd70000-10d7-11eb-97c4-4670ce7137f6.jpg)
I build my board to fit exactly between the clock and the original pcb.
The Wemos is places under the clock (right side of the photo).

I guess you would have to remove the clock.

**Comment by tom2199** [comment]
@mdaser did you try to build this project with the device in question? Did it work? An update would be awesome, thanks in advance.

**Comment by tom2199** [comment]
So I can confirm that this project works with this module (commonly known as Yunshan Relay Module, Tasmota even has built in support for it). @CWempe You are right the module is too big to fit without removing the clock, so it had to go. Below are a few pictures of my setup, I used the easiest solution I came up with, I just glued two layers of cardboard together and screwed the module down on it, the cardboard is held in place by the case.
![IMG_20210120_153605](https://user-images.githubusercontent.com/9537697/105206656-e8698580-5b46-11eb-9e8f-89ff651c6004.jpg) 
![IMG_20210120_154406](https://user-images.githubusercontent.com/9537697/105206821-14850680-5b47-11eb-9c4b-d5120ed882e4.jpg)
![IMG_20210120_154418](https://user-images.githubusercontent.com/9537697/105206862-2070c880-5b47-11eb-8c9f-271583f34903.jpg)
